core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -268,7 +268,8 @@ public BackupResponse backup(BackupPOptions options) throws IOException {
     String dir = options.hasTargetDirectory() ? options.getTargetDirectory()
         : ServerConfiguration.get(PropertyKey.MASTER_BACKUP_DIRECTORY);
     UnderFileSystem ufs = mUfs;
-    if (options.getLocalFileSystem() && !ufs.getUnderFSType().equals("local")) {
+    if ((options.getLocalFileSystem() && !ufs.getUnderFSType().equals("local"))
+            || !options.hasTargetDirectory()) {
```

Comment:
I think this condition should be
`((options.getLocalFileSystem() || !options.hasTargetDirectory) && !ufs.getUnderFSType().equals("local"))`

Reply:
done.